src/main/java/com/google/devtools/build/lib/skyframe/IgnoredPackagePrefixesFunction.java

```diff
@@ -44,9 +46,12 @@
  */
 public class IgnoredPackagePrefixesFunction implements SkyFunction {
```

Comment:
optional: the class name is technically no longer accurate, as this is more "ignored package prefixes and patterns function". Maybe just rename to "IgnoredPackagesFunction", although I understand that'll be a lot more changes across this PR, so up to you.

either way, please update the javadoc a bit to mention it also deals with glob patterns in REPO.bazel files (in addition to .bazelignore).

Reply:
If you don't mind, I don't do the rename now because it would indeed be too intrusive. If I don't forget, I'll do it in a followup change, though. I changed the Javadoc, though.